pkg/ceph/osd/agent.go

```diff
@@ -565,6 +565,13 @@ func refreshDeviceInfo(name string, nameToUUID map[string]string, scheme *PerfSc
 func (a *OsdAgent) startOSD(context *clusterd.Context, config *osdConfig) error {
 	newOSD := false
 	config.rootPath = path.Join(config.configRoot, fmt.Sprintf("osd%d", config.id))
+
+	if isFilestoreDevice(config) {
+		if err := remountFilestoreDeviceIfNeeded(context, config); err != nil {
```

Comment:
Should this check be placed in the `else` block line L623, since this is not a new OSD?